spec/system/dismiss_topics_spec.rb

```diff
@@ -0,0 +1,23 @@
+# frozen_string_literal: true
+
+describe "Filtering topics", type: :system, js: true do
+  fab!(:user) { Fabricate(:user) }
+  let(:topic_list) { PageObjects::Components::TopicList.new }
+  let(:dismiss_new_modal) { PageObjects::Modals::DismissNew.new }
+  fab!(:group) { Fabricate(:group).tap { |g| g.add(user) } }
+  fab!(:topic) { Fabricate(:topic) }
+
+  before { SiteSetting.experimental_new_new_view_groups = group.id }
+
+  it "displays confirmation modal with preselected options" do
+    sign_in(user)
+
+    visit("/new")
+
+    expect(topic_list).to have_topic(topic)
+    find(".dismiss-read").click
+    expect(dismiss_new_modal.dismiss_topics_checkbox).to be_checked
```

Comment:
I think we're still doing the same thing as what I mentioned previously where we're not allowing Capybara to wait until the field is checked. It seems like Capybara has a specific helper for this: https://www.rubydoc.info/gems/capybara/Capybara/Node/Matchers#has_checked_field%3F-instance_method